src/services/file/interfaces/fileRepository.ts

```diff
@@ -2,7 +2,11 @@ import { ReadStream } from 'fs';
 
 import { FastifyBaseLogger } from 'fastify';
 
+import { FileItemType } from '@graasp/sdk';
+
 export interface FileRepository {
+  getFileType(): FileItemType;
```

Comment:
What is the difference here ? The getter will generate the wanted function ?

Reply:
The big difference is how it is called.
```typescript
class A {
    public value: string = 'a';
}
console.log(new A().value);
```
```typescript
class B {
    public getValue() {return 'b';};
}
console.log(new B().getValue());
```
```typescript
class C {
    public get value() {return 'c';};
}
console.log(new C().value);
```

Currently we're using example B, but I'd recommend C. It would help refactoring between example A and C.
Also, when we're combining `getter` and `setter` function, the usage of the object is smoother. Instead of having that : 
```typescript
const bObject = new B();
bObject.setValue('b');
bObject.getValue();
```
We can have :
```typescript
const cObject = new C();
cObject.value = 'c';
cObject.value;
```

But the counterpart is that we won't be able to add parameters to the `getter` function, or we can't have a return value to `setter` function. So we should use it only when the wanted behavior match.
Probably, an another counterpart is that we're not expecting to call a function when accessing to a property

Reply:
I think you should not return on a setter and giving a params on a getter so it's not an issue in my point.

Reply:
I didn't know this syntax in typescript (finally very similar to C#), so thanks for your comment !